www/apps/book/app/learn/fundamentals/module-links/index-module/page.mdx

```diff
@@ -0,0 +1,487 @@
+export const metadata = {
+  title: `${pageNumber} Index Module`,
+}
+
+# {metadata.title}
+
+In this chapter, you'll learn about the Index Module and how you can use it.
+
+<Note type="soon" title="Experimental">
+
+The Index Module is an experimental feature that's subject to change. It is not recommended for production use.
+
+</Note>
+
+## What is the Index Module?
+
+The Index Module adds a method to [Query](../query/page.mdx) that allows you to filter data by linked modules.
+
+While modules share the same database by default, Medusa [isolates modules](../../modules/isolation/page.mdx) to allow using external data sources or different database types.
+
+So, when you retrieve data across modules using Query, Medusa aggregates the data coming from diffeent modules to create the end result. This approach limits your ability to filter data by linked modules. For example, you can't filter products (created in the Product Module) by their brand (created in the Brand Module).
+
+The Index Module solves this problem by injesting data into a central data store on application start up. The data store has a relational structure that enables efficiently filtering the data injested from different modules (and their data stores). So, when you retrieve data with the Index Module, you're retrieving them from the injested data store, not the original data source.
+
+![Diagram showcasing how data is retrieved from the Index Module's data store](https://res.cloudinary.com/dza7lstvk/image/upload/v1747988533/Medusa%20Book/index-module_epurmt.jpg)
+
+### Injested Data Models
+
+Currently, only the following data models are injested by the Index Module when you install it:
+
+- `Product`
+- `ProductVariant`
+- `Price`
+- `SalesChannel`
+
+Future versions may add more data models to the list.
+
+---
+
+## How to Install the Index Module
+
+To install the Index Module, run the following command in your Medusa project to install its package:
+
+```bash npm2yarn
+npm install @medusajs/index
+```
+
+Then, add the Index Module to your Medusa configuration in `medusa-config.ts`:
+
+```ts title="medusa-config.ts"
+module.exports = defineConfig({
+  // ...
+  modules: [
+    // ...
+    {
+      resolve: "@medusajs/index",
+    },
+  ],
+})
+```
+
+Finally, run the migrations to create the necessary tables for the Index Module in your database:
+
+```bash
+npx medusa db:migrate
+```
+
+The index module only injests data when you start your Medusa server. So, to injest the [currently supported data models](#injested-data-models), start the Medusa application:
```

Comment:
todo: This is not entirely true. There are two ways we ingest data:
1. We ingest data when we start the application, if the existing data in the index is stale
2. We ingest data whenever new entities (custom or core ones) are created, updated, or deleted
```suggestion
The index module only ingests data when you start your Medusa server. So, to injest the [currently supported data models](#injested-data-models), start the Medusa application:
```

Reply:
> We ingest data whenever new entities (custom or core ones) are created, updated, or deleted

Do you mean when we run migrations? Or when does that exactly happen?